src/kili/services/helpers.py

```diff
@@ -91,6 +89,35 @@ def infer_ids_from_external_ids(
     return id_map
 
 
+def _build_id_map(auth, asset_external_ids, project_id):
+    if len(asset_external_ids) > 1000:
+        assets_generators: List[Generator[Dict, None, None]] = []
+        # query all assets by external ids batches when there are too many
+        for external_ids_batch in pagination.BatchIteratorBuilder(
+            asset_external_ids, QUERY_BATCH_SIZE
+        ):
+            assets_generators.append(
+                AssetQuery(auth.client)(
+                    AssetWhere(project_id, external_id_contains=external_ids_batch),
+                    ["id", "externalId"],
+                    QueryOptions(disable_tqdm=True),
+                )
+            )
+        assets = chain(*assets_generators)
+    else:
+        # query all assets
+        assets = AssetQuery(auth.client)(
+            AssetWhere(project_id),
+            ["id", "externalId"],
+            QueryOptions(disable_tqdm=True),
+        )
```

Comment:
When `len(asset_external_ids) <= 1000`, we don't want to pass the `external_id_contains=asset_external_ids` in the AssetWhere? It's smaller than 1000 so the backend should not crash?

I wonder if we should just simply batch everything, regardless of the len(asset_external_ids)

Reply:
you're right, we can even batch with 1000 size

Reply:
changed